src/models/connectionStore.ts

```diff
@@ -293,13 +293,19 @@ export class ConnectionStore {
      */
     public removeRecentlyUsed(conn: IConnectionProfile): Promise<void> {
         const self = this;
-        return new Promise<void>((resolve, reject) => {
+        return new Promise<void>(async (resolve, reject) => {
             // Get all profiles
             let configValues = self.getRecentlyUsedConnections();
 
             // Remove the connection from the list if it already exists
             configValues = configValues.filter(value => !Utils.isSameProfile(<IConnectionProfile>value, conn));
 
+            // Remove any saved password
+            if (conn.savePassword) {
+                let credentialId = ConnectionStore.formatCredentialId(conn.server, conn.database, conn.user, ConnectionStore.CRED_MRU_USER);
```

Comment:
Do we really want to do that here? Removing something from the MRU doesn't necessarily mean we should remove the entire connection normally. I would expect that whatever is calling this should determine if the password needs to be cleared (such as if we're completely removing a connection from all the places it's stored such as the MRU and Connection Store)

Reply:
Fixed.